plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -837,34 +842,54 @@ private String toRemoteSchemaName(String schemaName)
         if (!caseInsensitiveNameMatching) {
             return schemaName;
         }
-        for (String remoteSchemaName : client.listDatabaseNames()) {
+        for (String remoteSchemaName : listDatabaseNames()) {
             if (schemaName.equals(remoteSchemaName.toLowerCase(ENGLISH))) {
                 return remoteSchemaName;
             }
         }
         return schemaName;
     }
 
+    private Iterable<String> listDatabaseNames()
```

Comment:
Maybe instead returning `Iterable` from here and later turning it to Stream through `ImmutableList.of(listDatabaseNames()).stream()`, please return `Iterable` from here, for example by `Streams.stream(...)`.

Reply:
We discuss this offline, because `listDatabaseNames` is private I return now `MongoIterable`